clients/go/admin/client_test.go

```diff
@@ -55,9 +59,11 @@ func TestInitializeMockAdminClient(t *testing.T) {
 func TestGetAdditionalAdminClientConfigOptions(t *testing.T) {
```

Comment:
Do we require a unit test where the config is initialised from a file.
This is how flytectl config is defined .

```
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///flyte.myexample.com
  authType: Pkce
logger:
  show-source: true
  level: 1
```
Wanted to make sure endpoint and authType are read correctly with this refactoring into BaseConfig.